src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -150,30 +150,72 @@ static class Extensions
 	{
 		private const uint Wgs84SpatialReferenceId = 4326;
 
-		public static Geoposition ToGeoPosition(this Location location)
-			=> new Geoposition(
-				new Geocoordinate(
-					latitude: location.Latitude,
-					longitude: location.Longitude,
-					altitude: location.Altitude,
-					timestamp: FromUnixTime(location.Time),
-					speed: location.HasSpeed ? location.Speed : 0,
-					point: new Geopoint(
-						new BasicGeoposition
-						{
-							Latitude = location.Latitude,
-							Longitude = location.Longitude,
-							Altitude = location.Altitude,
-						},
-						AltitudeReferenceSystem.Ellipsoid,
-						Wgs84SpatialReferenceId
-					),
-					accuracy: 0,
-					altitudeAccuracy: 0,
-					heading: null
-				)
-			);
-
+		public static Windows.Devices.Geolocation.Geoposition ToGeoPosition(this Location location)
+		{
+			double? geoheading = null;
+			if (location.HasBearing)
+			{
+				geoheading = location.Bearing;
+			}
+
+			Windows.Devices.Geolocation.PositionSource posSource;

```

Comment:
You could use the new 'switch expression syntax here https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression

Reply:
for me, this syntax is less readable :) maybe simple it is too new...